contracts/gateway/EIP20Gateway.sol

```diff
@@ -617,21 +615,21 @@ contract EIP20Gateway is GatewayBase {
         staker_ = message.sender;
         stakerNonce_ = message.nonce;
         amount_ = stakes[_messageHash].amount;
+        uint256 stakeBounty = stakes[_messageHash].bounty;
 
-        // transfer the staked amount to the staker
+        delete stakes[_messageHash];
+
+        // Transfer the staked amount to the staker.
         token.transfer(message.sender, amount_);
 
-        // burn facilitator bounty
-        baseToken.transfer(burner, bounty);
-        //penalty charged to staker
-        uint256 penalty = penaltyFromBounty(stakes[_messageHash].bounty);
+        // Burn facilitator bounty.
+        baseToken.transfer(burner, stakeBounty);
+        // Penalty charged to staker.
+        uint256 penalty = penaltyFromBounty(stakeBounty);
```

Comment:
Will `penalty` always be zero for a zero bounty? If we cannot guarantee that, the current change could still lead to problems for a zero `stakeBounty` (after deleting the stake from the mapping and reentrying).

Reply:
Reentrancy will be handled by `progressOutboxRevocation` on L606. What do you suggest?
Somethin like this?
require(
            stakes[_messageHash].amount > 0,
            "Stake amount must be greater than zero."
)